pkg/config/mesh/networks_watcher.go

```diff
@@ -46,17 +43,17 @@ type NetworksWatcher interface {
 	NetworksHolder
 
 	// AddNetworksHandler registers a callback handler for changes to the networks config.
-	AddNetworksHandler(func()) WatcherHandlerRegistration
+	AddNetworksHandler(func()) *WatcherHandlerRegistration
 
 	// DeleteNetworksHandler unregisters a callback handler when remote cluster is removed.
-	DeleteNetworksHandler(registration WatcherHandlerRegistration)
+	DeleteNetworksHandler(registration *WatcherHandlerRegistration)
```

Comment:
If I was designing it I would have 

```go
type Handler interface {
  Unsubscribe()
}
AddNetworksHandler(func()) Handler
```

Then they call
```go
handler := AddNetworksHandler(func(){})
handler.Unsubscribe()
```

Reply:
This works fine as  well though

Reply:
```
type Handler interface {
  Unsubscribe()
}
```

Sounds good